polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -75,124 +75,43 @@ MLIRScanner::MLIRScanner(MLIRASTConsumer &Glob,
                          LowerToInfo &LTInfo)
     : Glob(Glob), function(), module(module), builder(module->getContext()),
       loc(builder.getUnknownLoc()), entryBlock(nullptr), loops(),
-      allocationScope(nullptr), supportedFuncs(), bufs(), constants(), labels(),
-      EmittingFunctionDecl(nullptr), params(), Captures(), CaptureKinds(),
-      ThisCapture(nullptr), arrayinit(), ThisVal(), returnVal(),
+      allocationScope(nullptr), unsupportedFuncs(), bufs(), constants(),
+      labels(), EmittingFunctionDecl(nullptr), params(), Captures(),
+      CaptureKinds(), ThisCapture(nullptr), arrayinit(), ThisVal(), returnVal(),
       LTInfo(LTInfo) {}
 
-void MLIRScanner::initSupportedFunctions() {
-  // Functions needed for single_task with one dimensional write buffer.
-
-  // SYCL constructors:
-  supportedFuncs.insert("_ZN4sycl3_V16detail18AccessorImplDeviceILi1EEC1ENS0_"
-                        "2idILi1EEENS0_5rangeILi1EEES7_");
-  supportedFuncs.insert(
-      "_ZN4sycl3_V18accessorIiLi1ELNS0_6access4modeE1025ELNS2_"
-      "6targetE2014ELNS2_11placeholderE0ENS0_"
-      "3ext6oneapi22accessor_property_listIJEEEEC1Ev");
-  supportedFuncs.insert("_ZN4sycl3_V16detail5arrayILi1EEC1ILi1EEENSt9enable_"
-                        "ifIXeqT_Li1EEmE4typeE");
-  supportedFuncs.insert("_ZN4sycl3_V16detail5arrayILi1EEC1ERKS3_");
-  supportedFuncs.insert("_ZN4sycl3_V12idILi1EEC1Ev");
-  supportedFuncs.insert("_ZN4sycl3_V12idILi1EEC1ERKS2_");
-  supportedFuncs.insert(
-      "_ZN4sycl3_V12idILi1EEC1ILi1EEENSt9enable_ifIXeqT_Li1EEmE4typeE");
-  supportedFuncs.insert("_ZN4sycl3_V15rangeILi1EEC1ERKS2_");
-  supportedFuncs.insert(
-      "_ZN4sycl3_V15rangeILi1EEC1ILi1EEENSt9enable_ifIXeqT_Li1EEmE4typeE");
-
-  // Other SYCL functions:
-  supportedFuncs.insert(
-      "_ZN4sycl3_V13ext6oneapi22accessor_property_listIJEE12has_propertyINS2_"
-      "8property9no_offsetEEEbPNSt9enable_ifIXsr24is_compile_time_propertyIT_"
-      "EE5valueEvE4typeE");
-  supportedFuncs.insert(
-      "_ZNK4sycl3_V18accessorIiLi1ELNS0_6access4modeE1025ELNS2_"
+void MLIRScanner::initUnsupportedFunctions() {
+  // FIXME: -no-mangled-function-name: cgeist:
```

Comment:
@victor-eds Can you help look into this `-no-mangled-function-name` issue?

Reply:
I'll look into this. I think we can merge this one and remove this from the list when the problem is solved.